onc-healthit/lantern-back-end
# Add queues

Changes in this PR:
- The `endpointmanager` now creates two queues, one for the `capabilityquerier` and one for the `networkstatsquerier`, and sends the endpoint information from the database to both queues on given intervals
- Moved the `ConnectToQueue` function and its test from the `capabilityquerier` into the `lanternmq` package, and added some other related functions, so that these functions could be used in the other packages.
- Renamed test queue environment variable
- Added a new `GetAllFHIREndpoints` function
- Generalized the Topic functions in `messagequeue.go` so that it could work with any exchange type (ultimately not used)

**To Test**:
- Run in the `endpointmanager/cmd/sendendpoints/` directory: `go run main.go`
- If you log into rabbitmq, you should see that 2 new queues have been created (`endpoints-to-capability` and `endpoints-to-netstats`) and have 397 messages in each
- Change either environment variable interval (`LANTERN_CAPQUERY_QRYINTVL` or `LANTERN_ENDPTQRY_QUERY_INTERVAL`) to a shorter period of time and rerun the `endpointmanager` to see the messages get added to the queue on that time interval.
--------------
Pull requests into this repository require the following checks to be completed by the submitter and reviewers.

**Submitter:**
  - This pull request describes why these changes were made.
  - The JIRA ticket links to this PR.
  - The JIRA ticket for this PR is: https://oncprojectracking.healthit.gov/support/browse/LANTERN-160
  - Tests are included and test edge cases.
  - Tests have been run locally and pass.
  - Code diff has been done and checked to ensure only expected code is being committed.
  - A linter for the language has been run.

**Primary Reviewer:**

  - Code has been run locally and the basic functionality affected by the PR has been checked.
  - Edge cases have been checked to ensure behavior is as expected.
  - Code is maintainable, reusable, efficient, and correct.
  - Code accomplishes the tasks purpose.
  - Code follows style guidance appropriate for the language, including passing linter checks.
  - Code is well commented (which does not mean verbosely commented).
  - Tests are included and test edge cases.
  - Tests have been run locally and pass.

**Secondary Reviewer:**

  - Code has been run locally and the basic functionality affected by the PR has been checked.
  - Edge cases have been checked to ensure behavior is as expected.
  - Code accomplishes the tasks purpose.
  - Tests have been run locally and pass.

**Merger:**

Prior to merging, ensure that all checks have been completed. Delete the branch after the merge
is complete.

Create a new branch `<branch name>_gomodupdate`. Run `make update_mods` to update the `go.mod`
and `go.sum` files to point to master. Create a PR. Require one sanity check reviewer.
